x/profile/internal/keeper/keeper.go

```diff
@@ -0,0 +1,135 @@
+package keeper
+
+import (
+	"bytes"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/desmos-labs/desmos/x/profile/internal/types"
+)
+
+// Keeper maintains the link to data storage and exposes getter/setter methods for the various parts of the state machine
+type Keeper struct {
+	StoreKey sdk.StoreKey // Unexposed key to access store from sdk.Context
+	Cdc      *codec.Codec // The wire codec for binary encoding/decoding.
+}
+
+// NewKeeper creates new instances of the magpie Keeper
+func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey) Keeper {
+	return Keeper{
+		StoreKey: storeKey,
+		Cdc:      cdc,
+	}
+}
+
+// AssociateMonikerWithAddress save the relation of moniker and address on chain
+func (k Keeper) AssociateMonikerWithAddress(ctx sdk.Context, moniker string, address sdk.AccAddress) {
+	store := ctx.KVStore(k.StoreKey)
+	key := types.MonikerStoreKey(moniker)
+	store.Set(key, k.Cdc.MustMarshalBinaryBare(&address))
+}
+
+// GetMonikerRelatedAddress returns the address associated to the given moniker or nil if it not exists
+func (k Keeper) GetMonikerRelatedAddress(ctx sdk.Context, moniker string) (addr sdk.AccAddress) {
+	store := ctx.KVStore(k.StoreKey)
+
+	key := types.MonikerStoreKey(moniker)
+	bz := store.Get(key)
+	if bz == nil {
+		return nil
+	}
+
+	k.Cdc.MustUnmarshalBinaryBare(bz, &addr)
+
+	return addr
```

Comment:
We can make this more compact

```suggestion
	store := ctx.KVStore(k.StoreKey)
	bz := store.Get(types.MonikerStoreKey(moniker))
	if bz == nil {
		return nil
	}

	k.Cdc.MustUnmarshalBinaryBare(bz, &addr)
	return addr
```